mesonbuild/compilers/d.py

```diff
@@ -341,14 +341,25 @@ def _translate_args_to_nongnu(self, args: T.List[str]) -> T.List[str]:
                 # Hence, we guess here whether the flag was intended to pass
                 # a linker search path.
 
+                if arg.startswith('-L='):
+                    suffix = arg[3:]
+                else:
+                    suffix = arg[2:]
+
+                # The user has passed a linker flag directly for the D compiler
+                if suffix.startswith('-L'):
+                    dcargs.append(arg)
+                    continue
+
+                # linker flag such as -L=/DEBUG must pass through
+                if self.linker.id == 'link' and self.info.is_windows() and suffix.startswith('/'):
+                    dcargs.append(arg)
+                    continue
```

Comment:
This function actually receives flag such as `-L=/DEBUG` `-L=/SUBSYSTEM:CONSOLE` etc.
They are correctly passed to the linker only because of the '=' sign, not because the logic is correct.